source/channel.c

```diff
@@ -646,7 +646,12 @@ void aws_channel_schedule_task_future(
 }
 
 bool aws_channel_thread_is_callers_thread(struct aws_channel *channel) {
-    return aws_event_loop_thread_is_callers_thread(channel->loop);
+    // As aws_event_loop is not ref-counted, it is possible that the socket read/write callback get triggered after
+    // the event loop is released, ignore the function call here.
+    if (channel && channel->loop) {
```

Comment:
This is alarming.  I would like to know a lot more about how this is possible: how can you have a channel still alive while its event loop is not?